jib-core/src/main/java/com/google/cloud/tools/jib/frontend/JavaLayerConfigurations.java

```diff
@@ -36,47 +36,62 @@
   /** Represents the different types of layers for a Java application. */
   @VisibleForTesting
   enum LayerType {
-    DEPENDENCIES("dependencies", JavaEntrypointConstructor.DEFAULT_DEPENDENCIES_PATH_ON_IMAGE),
+    DEPENDENCIES(
```

Comment:
Thinking more about this enum where we need to dynamically decide the final extraction path, I'm now leaning toward getting rid of static extraction path initialization with the `appRootRelative` switch. Instead, the `JavaLayerConfiguration.Builder()` could accept the extraction path when setting layer files. That is, change `setClassFiles(List<Path> classFiles)` to `setClassFiles(String extractionPath, List<Path> classFiles)`. That seems more straightforward for our dynamic path requirement and simpler. What do you think?

Reply:
Yea that sounds good. One nit is to have it be `setClassFiles(List<Path> classFiles, String extractionPath)` to be more like `LayerEntry`.